dotcom-rendering/src/components/SubheadingBlockComponent.stories.tsx

```diff
@@ -0,0 +1,133 @@
+import { css } from '@emotion/react';
+import { ArticleDesign, ArticleDisplay, Pillar } from '@guardian/libs';
+import { headline, textSans } from '@guardian/source-foundations';
+import { type Decorator } from '@storybook/react';
+import { splitTheme } from '../../.storybook/decorators/splitThemeDecorator';
+import { SubheadingBlockComponent } from './SubheadingBlockComponent';
+
+const standardFormat = {
+	design: ArticleDesign.Standard,
+	display: ArticleDisplay.Standard,
+	theme: Pillar.News,
+};
+
+const immersiveFormat = {
+	...standardFormat,
+	display: ArticleDisplay.Immersive,
+};
+
+/** Mocking the styles normally inherited via ArticleBody component */
+const GlobalStylesDecorator =
+	(format: ArticleFormat): Decorator =>
+	(Story) => (
+		<div
+			css={css`
+				h2:not([data-ignore='global-h2-styling']) {
+					${format.display === ArticleDisplay.Immersive
+						? headline.medium({ fontWeight: 'light' })
+						: headline.xxsmall({ fontWeight: 'bold' })};
+				}
+			`}
+		>
+			{Story()}
+		</div>
+	);
+
+const StoryWrapper = ({ children }: { children: React.ReactNode }) => (
+	<div
+		css={css`
+			span {
+				${textSans.medium({ fontStyle: 'italic' })};
+				margin-bottom: 4px;
+			}
+			hr {
+				border: 0.5px dashed;
+			}
+			margin: 10px;
+
+			* {
+				margin-bottom: 10px;
+			}
+		`}
+	>
+		{children}
+	</div>
+);
+
+export default {
+	component: SubheadingBlockComponent,
+	title: 'Components/SubheadingBlockComponent',
+	render: (args: React.ComponentProps<typeof SubheadingBlockComponent>) => {
```

Comment:
If you do:

```ts
const meta = {
  component: SubheadingBlockComponent,
  ...
} satisfies Meta<typeof SubheadingBlockComponent>;

export default meta;
```

you should get automatic type safety on the `args` parameter to the `render` function.

This also allows you to do:

```ts
type Story = StoryObj<typeof meta>;

export const StandardDisplay = {
  ...
} satisfies Story;
```

to get better type safety on the stories themselves. See #10880 for an example.